apps/andi/lib/andi_web/live/edit_live_view.ex

```diff
@@ -1,178 +1,179 @@
+          <%= text_input(f, :sourceFormat, [class: "input--text input", readonly: true]) %>
+          <%= error_tag(f, :sourceFormat) %>
+        </div>
+        <div class="metadata-form__level-of-access">
+          <%= label(f, :private, "Level of Access", class: "label label--required") %>
+          <%= select(f, :private, get_level_of_access_options(), class: "select") %>
+          <%= error_tag(f, :private) %>
+        </div>
         <div class="metadata-form__cancel-btn">
           <%= Link.button("Cancel", to: "/", method: "get", class: "btn btn--cancel") %>
         </div>
         <div class="metadata-form__save-btn">
-          <%= Link.button("Next", to: "/", method: "get", id: "next-button", class: "btn btn--next") %>
-          <%= Form.submit("Save", id: "save-button", class: "btn btn--save") %>
+          <%= unless is_nil(@validation_errors) do %>
+            <div class="metadata__error-message">
+              <span>There were errors with the dataset you tried to submit.
+              <ul>
+                <%= for error <- @validation_errors do %>
+                  <li><%= error %></li>
+                <% end %>
+              </ul>
+            </div>
+          <% end %>
+          <%= Link.button("Next", to: "/", method: "get", id: "next-button", class: "btn btn--next", disabled: true, title: "Not implemented yet.") %>
+          <%= submit("Save", id: "save-button", class: "btn btn--save") %>
         </div>
       </div>
-      <div>
-        <%= if @is_saved do %>
-          <div id="success-message" class="metadata__success-message">Saved Successfully</div>
-        <% end %>
-      </div>
+      <%= if @save_success do %>
+        <div id="success-message" class="metadata__success-message">Saved Successfully</div>
+      <% end %>
 
     """
   end
 
   def mount(%{dataset: dataset}, socket) do
-    {:ok,
-     assign(socket, id: dataset.id, dataset: dataset, changeset: Andi.DatasetSchema.changeset(dataset), is_saved: false)}
-  end
+    new_changeset = Metadata.changeset_from_struct(dataset)
+
+    socket =
+      assign(socket,
+        dataset: dataset,
+        changeset: new_changeset,
+        validation_errors: nil,
+        save_success: false
+      )
 
-  def handle_event(
-        "validate",
-        %{"dataset_schema" => dataset_schema},
-        socket
-      ) do
-    {:noreply, assign(socket, changeset: apply_changes(dataset_schema), is_saved: false)}
+    {:ok, socket}
   end
 
-  def handle_event("save", %{"dataset_schema" => dataset_schema}, socket) do
-    change = apply_changes(dataset_schema)
+  def handle_event("validate", %{"metadata" => form_data}, socket) do
+    socket = reset_save_success(socket)
 
-    case change.valid? do
-      true ->
-        schema = Ecto.Changeset.apply_changes(change)
-        original_dataset = socket.assigns.dataset
+    new_changeset =
+      form_data
+      |> Metadata.new_changeset()
+      |> Map.put(:action, :update)
```

Comment:
What does changing the action here do?

Reply:
Putting an action onto the changeset will cause the `Phoenix.HTML` code to show errors on the form. This keeps a whole page of errors from showing up before the user has even edited a field. Not having an action on the changeset is the reason the code had to access `form.source.errors` on the `ErrorHelper` module.